staging/src/k8s.io/kube-openapi/pkg/aggregator/aggregator.go

```diff
@@ -213,10 +209,29 @@ func equalSchemaMap(s1, s2 map[string]spec.Schema) bool {
 	return true
 }
 
-func equalSchemaArray(s1, s2 []spec.Schema) bool {
-	if s1 == nil || s2 == nil {
-		return s1 == nil && s2 == nil
+func toSchemaArrayWithUniqueItems(in []spec.Schema) []spec.Schema {
+	out := []spec.Schema{}
+	if in == nil || len(in) == 0 {
+		return out
 	}
+	for _, v1 := range in {
+		found := false
+		for _, v2 := range out {
+			if EqualSchema(&v1, &v2) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			out = append(out, v1)
+		}
+	}
+	return out
+}
+
+func equalSchemaArray(in1, in2 []spec.Schema) bool {
+	s1 := toSchemaArrayWithUniqueItems(in1)
```

Comment:
compare my `oneOf` example. At least for that construct we must not unify. I guess for all other JSON-Schema constructs that's fine.